sdk/dotnet/Pulumi/Core/Input.cs

```diff
@@ -21,29 +23,34 @@ internal interface IInput
     /// </summary>
     public class Input<T> : IInput
     {
-        /// <summary>
-        /// Technically, in .net we can represent Inputs entirely using the Output type (since
-        /// Outputs can wrap values and promises).  However, it would look very weird to state that
-        /// the inputs to a resource *had* to be Outputs. So we basically just come up with this
-        /// wrapper type so things look sensible, even though under the covers we implement things
-        /// using the exact same type
-        /// </summary>
-        private protected Output<T> _outputValue;
+        private readonly OneOf<T, Output<T>> _value;
```

Comment:
Ok so this is readonly means that InputList completely ignores this slot right, it just has its own slot called _value and overrides the accessors.. Sorry I have not worked much with inheritance and struggling a bit here.

Reply:
Yes, that's right. It's unfortunate that this slot (field) can't be used by `InputList<T>` (or the map), but I have to do something like this since `InputList<T>` is `Input<ImmutableArray<T>>` rather than `Input<ImmutableArray<Input<T>>>` as you pointed out. Since the subclasses aren't using the field anymore, they don't need access to it, so I've tightened up the visibility and made it readonly.